numpy/random/mtrand/mtrand.pyx

```diff
@@ -1292,6 +1293,12 @@ cdef class RandomState:
                                   fscale, self.lock)
 
         temp = np.subtract(ohigh, olow)
+
+        for i in range(temp.shape[0]):
+            fscale = PyFloat_AsDouble(temp[i])
+            if not npy_isfinite(fscale):
+                raise OverflowError('Range exceeds valid bounds')
+
         Py_INCREF(temp)  # needed to get around Pyrex's automatic reference-counting
                          # rules because EnsureArray steals a reference
         odiff = <ndarray>PyArray_EnsureArray(temp)
```

Comment:
You could just use `not np.any(np.isfinite(odiff))` here rather than manually looping. This is used multiple places, e.g.,

https://github.com/alimuldal/numpy/blob/341ed78f06e100c921de46b047fca8d7f77d5fca/numpy/random/mtrand/mtrand.pyx#L1642